src/internal/proc.js

```diff
@@ -11,16 +11,17 @@ export const TASK_CANCEL = {toString() { return '@@redux-saga/TASK_CANCEL' }}
 
 const matchers = {
   wildcard  : () => kTrue,
-  default   : pattern => input => input.type === pattern,
-  array     : patterns => input => patterns.some(p => p === input.type),
+  default   : pattern => input => input.type === String(pattern),
+  array     : patterns => input => patterns.some(p => String(p) === input.type),
   predicate : predicate => input => predicate(input)
 }
 
 function matcher(pattern) {
   return (
-      pattern === '*'   ? matchers.wildcard
-    : is.array(pattern) ? matchers.array
-    : is.func(pattern)  ? matchers.predicate
+      pattern === '*'            ? matchers.wildcard
+    : is.array(pattern)          ? matchers.array
+    : is.stringableFunc(pattern) ? matchers.default
```

Comment:
still wondering if this should be restricted to just stringable functions, although when I wonder about this:
```javascript
{
  type: 'TYPE_A',
  toString: () => 'TYPE_B',
}
```
I have cold chills, so maybe only functions after all, WDYT? do u know about any other libs with different approach to actions than basic one?